python/cudf/cudf/core/column/lists.py

```diff
@@ -278,6 +279,48 @@ def leaves(self):
         else:
             return self.elements
 
+    @classmethod
+    def from_sequences(
+        cls, arbitrary: Sequence[ColumnLike]
+    ) -> "cudf.core.column.ListColumn":
+        """
+        Create a list column for list of column-like sequences
+        """
+        data_col = column.column_empty(0)
+        mask_col = []
+        lengths_col = []
+
+        # Build Data & Mask
+        for data in arbitrary:
+            if cudf._lib.scalar._is_null_host_scalar(data):
+                mask_col.append(False)
+                lengths_col.append(0)
+            else:
+                mask_col.append(True)
+                data_col = data_col.append(as_column(data))
+                lengths_col.append(len(data))
+
+        # Build offsets
+        offset_col = column.column_empty(
+            row_count=len(arbitrary) + 1, dtype="int32"
+        )
+        offset_col[0] = 0
+        offset_col[1:] = lengths_col
+        offset_col = cast(
+            cudf.core.column.NumericalColumn, offset_col
+        )._apply_scan_op("sum")
```

Comment:
Not sure what the overhead on `_apply_scan_op()` is, but would it make more sense to do this cumsum operation during the construction of `lengths_col` rather than here? Something like

```python

        # Build Data & Mask
        offset = 0
        for data in arbitrary:
            if cudf._lib.scalar._is_null_host_scalar(data):
                mask_col.append(False)
                lengths_col.append(offset)
            else:
                mask_col.append(True)
                data_col = data_col.append(as_column(data))
                offset += len(data)
                lengths_col.append(offset)
```

Reply:
Yup, seems like we have a very minimal optimization here:


```python
Using your approach:

In [1]: import cudf

In [2]: s = cudf.Series([1, 10, None, 345])

In [3]: %timeit cudf.Series([None, s, None, s, s, None, s])
2.79 ms ± 32 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

In [4]: %timeit cudf.Series([None, s, None, s, s, None, s]*100)
59.5 ms ± 1.89 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)

In [5]: %timeit cudf.Series([None, s, None, s, s, None, s]*1000)
566 ms ± 6.02 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)



Using _apply_scan_op:

In [1]: import cudf

In [2]: s = cudf.Series([1, 10, None, 345])

In [3]: %timeit cudf.Series([None, s, None, s, s, None, s])
4.34 ms ± 458 µs per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [4]: %timeit cudf.Series([None, s, None, s, s, None, s]*100)
60.6 ms ± 524 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)

In [5]: %timeit cudf.Series([None, s, None, s, s, None, s]*1000)
582 ms ± 8.84 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)



```